From 94516ebcf10eb8e9395e656fe7a241e16ed37ef5 Mon Sep 17 00:00:00 2001
From: Alaa Hleihel <ahleihel@redhat.com>
Date: Mon, 1 Jun 2020 15:40:46 -0400
Subject: [PATCH 293/312] [netdrv] net/mlx5: DR: Fix incorrect type in argument

Message-id: <20200601154102.25980-24-ahleihel@redhat.com>
Patchwork-id: 315727
Patchwork-instance: patchwork
O-Subject: [RHEL8.3 BZ 1842258 23/39] net/mlx5: DR: Fix incorrect type in argument
Bugzilla: 1842258
RH-Acked-by: Honggang Li <honli@redhat.com>
RH-Acked-by: Kamal Heib <kheib@redhat.com>
RH-Acked-by: Marcelo Leitner <mleitner@redhat.com>
RH-Acked-by: Jarod Wilson <jarod@redhat.com>

Bugzilla: http://bugzilla.redhat.com/1842258
Upstream: git://git.kernel.org/pub/scm/linux/kernel/git/davem/net-next.git , branch: master

commit 618f88c4c40a9621a3105f3ff957a91a148e7d94
Author: Saeed Mahameed <saeedm@mellanox.com>
Date:   Thu May 28 01:02:08 2020 -0700

    net/mlx5: DR: Fix incorrect type in argument

    HW spec objects should receive a void ptr to work on, the MLX5_SET/GET
    macro will know how to handle it.

    No need to provide explicit or wrong pointer type in this case.

    warning: incorrect type in argument 1 (different base types)
        expected unsigned long long const [usertype] *sw_action
        got restricted __be64 [usertype] *[assigned] sw_action

    Signed-off-by: Saeed Mahameed <saeedm@mellanox.com>
    Reviewed-by: Mark Bloch <markb@mellanox.com>

Signed-off-by: Alaa Hleihel <ahleihel@redhat.com>
Signed-off-by: Frantisek Hrbata <fhrbata@redhat.com>
---
 drivers/net/ethernet/mellanox/mlx5/core/steering/dr_action.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/mellanox/mlx5/core/steering/dr_action.c b/drivers/net/ethernet/mellanox/mlx5/core/steering/dr_action.c
index 1d90378b155c..c3cc01f93672 100644
--- a/drivers/net/ethernet/mellanox/mlx5/core/steering/dr_action.c
+++ b/drivers/net/ethernet/mellanox/mlx5/core/steering/dr_action.c
@@ -1644,7 +1644,7 @@ dr_action_modify_check_field_limitation(struct mlx5dr_action *action,
 }
 
 static bool
-dr_action_modify_check_is_ttl_modify(const u64 *sw_action)
+dr_action_modify_check_is_ttl_modify(const void *sw_action)
 {
 	u16 sw_field = MLX5_GET(set_action_in, sw_action, field);
 
-- 
2.13.6

